pkg/sypgp/sypgp.go

```diff
@@ -445,17 +453,19 @@ func (keyring *Handle) RemovePubKey(toDelete string) error {
 	return keyring.storePubKeyring(newKeyList)
 }
 
-func (keyring *Handle) genKeyPair(name, comment, email, passphrase string) (*openpgp.Entity, error) {
+func (keyring *Handle) genKeyPair(opts GenKeyPairOptions) (*openpgp.Entity, error) {
 	conf := &packet.Config{RSABits: 4096, DefaultHash: crypto.SHA384}
 
-	entity, err := openpgp.NewEntity(name, comment, email, conf)
+	entity, err := openpgp.NewEntity(*opts.Name, *opts.Comment, *opts.Email, conf)
```

Comment:
Possible nil dereference?

Reply:
Nevermind.
Maybe add a comment clarifying that `opts` should not container nil values (except pswd)?